### ISSUE TITLE
Docs need an API section

### ISSUE THREAD

**Issue by tomwayson** [Esri/angular-esri-map#99]
Need to document API of every service/directive.

May want to wait until after #69

**Comment by jwasil** [comment]
As you mention in #69, it would be great to
>  enable the "roll your own custom map directive" pattern

We've discussed this elsewhere, but I wanted to document here, too, that it would be great to lay out some kind of step-by-step rough guide on making custom map directives in a repeatable pattern.  Perhaps those who are trying to figure out the whole Angular and Esri JSAPI 3.x integration might benefit from that.

**Comment by tomwayson** [comment]
I've also been thinking of adding a "Patterns" section or the API for a "roll your own" example as well as the lazy loading example.

**Comment by tomwayson** [comment]
This is a reminder to us to add nav links to ngdoc header via:

https://github.com/nikhilmodak/gulp-ngdocs#navtemplate

**Comment by jwasil** [comment]
Add comments for `ngdoc`-umentation API generation to:
- [ ] map/esriMap
- [x] map/esriLegend
- [ ] layers/esriFeatureLayer
- [ ] layers/esriDynamicMapServiceLayer
- [ ] layers/esriInfoTemplate
- [ ] core/esriLayerUtils (_?_)
- [ ] core/esriLoader (_some documentation already started_)
- [ ] core/esriMapUtils (_?_)
- [ ] core/esriRegistry

Do we want to document the **xyzController.js** controllers and **xyzUtils.js** services?  Assuming yes for latter, not sure about former.

**Comment by tomwayson** [comment]
Yep, doc utils and controllers.

**Comment by jwasil** [comment]
Working on this in **api-documentation** branch.

**Comment by tomwayson** [comment]
FYI - core/esriLayerUtils is in the karma branch b/c I had to make some changes to that file while working on the unit tests (unit tests caught some bugs!).

**Comment by tomwayson** [comment]
@jwasil 

Why don't you PR what you have so far. I want to tag a Beta 6 release ASAP. We'll carry on the remaining ones under RC1 (maybe under a new issue like #146).

**Comment by tomwayson** [comment]
@jwasil I'm going to handle core/esriMapUtils as I'll be updating the actual api in #144.

**Comment by jwasil** [comment]
Great! I'm hoping to get on some of this later today, but feel free to jump on this branch if you need to for anything else.

**Comment by jwasil** [comment]
Continuing on branch **api-docs-continued**

**Comment by tomwayson** [comment]
:+1:

**Comment by jwasil** [comment]
@tomwayson would you be down with giving some of the docs a quick overview?  Some descriptions and params I wasn't completely sure about, and I just stood up the EsriLayerControllerBase and also not sure if/how we want to document `_bindLayerEvents`.  ngdoc doesn't seem to care about any `@private` flags, fyi.

We're almost done with these!

**Comment by tomwayson** [comment]
great work @jwasil! I was thinking that this documentation effort was going to be be mainly copying and pasting the existing comments into ngdocs format, but after doing esriMapUtils I realized how much can be added, and I think what you've done is looking great.

If we're going to doc the layer controllers that inherit from EsriLayerControllerBase, can we just ngdoc the `bindLayerEvents()` in each instead of `_bindLayerEvents()`? I'll defer to you on that.

**Comment by jwasil** [comment]
Thanks for the review and feedback!  Agreed and done for the `bindLayerEvents()` suggestion.
